src/Compression/Utilities.cpp

```diff
@@ -0,0 +1,15 @@
+#include "Utilities.h"
+
+namespace DB
+{
+CityHash_v1_0_2::uint128 CalculateCityHash128InLittleEndian(const std::span<char> buffer)
+{
+    const auto hash = CityHash_v1_0_2::CityHash128(buffer.data(), buffer.size());
+
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
This is inconsistent with the other parts of the code base.
See `std::endian`

Reply:
I would've preferred to do it this way:
```
if constexpr (std::endian::native == std::endian::big)
```
but this approach results in a build error (`error: code will never be executed [-Werror,-Wunreachable-code]`). Disabling the warnings for this scope would also have been an option, but that's also horrible. What would you recommend?